lib/src/Pos/Launcher/Launcher.hs

```diff
@@ -58,9 +60,22 @@ launchNode
     -> IO ()
 launchNode nArgs cArgs lArgs action = do
     let confOpts = configurationOptions (commonArgs cArgs)
-    let confKey = cfoKey confOpts
-    let withLogger' = loggerBracket confKey lArgs . logException (lpDefaultName lArgs)
-    let withConfigurations' = withConfigurations
+        confKey = cfoKey confOpts
+        withLogger' = loggerBracket confKey lArgs . logException (lpDefaultName lArgs)
+        -- FIXME
+        -- There is a `Wlog.WithLogger` instance on `IO` defined in
+        -- `Pos.Util.Wlog.Compatibility`. We can get a `Trace` from it, but
+        -- it will `error` if used before logging is initialized by
+        -- `withLogger'`. I cannot complain enough about this. Somebody chose
+        -- global uninitialized mutable state, just because they wanted to be
+        -- able to magically log in any `IO`. But the use of `error`s in that
+        -- instance makes it obvious that it's all a lie: you can't log anytime
+        -- anywhere, so why claim that you can? Just to avoid passing a
+        -- parameter? To "write less code"?
+        traceInfo :: Trace IO Text
+        traceInfo = contramap ((,) Info) fromTypeclassWlog
```

Comment:
You can use https://github.com/input-output-hk/cardano-sl/blob/develop/util/src/Pos/Util/Trace.hs#L101-L102.

Reply:
Or we can change `logInfo` to be:
```
logInfo :: Text -> Trace IO (Log.Severity, Text) -> IO ()
logInfo msg tr = traceWith (contramap ((,) Info) tr) msg

Reply:
Or equivalently

```
logInfo :: Trace IO (Log.Severity, Text) -> Text -> IO ()
logInfo trace = contramap ((,) Info) (traceWith trace)
```

Reply:
Yes that is better. Should I push a relevant commit?

Reply:
If you want to. But if we change `logInfo` there, we should change it for `logDebug`, `logError`, etc. If this causes a fallout of many other required changes then I'd say don't bother.

I tend to like using `traceWith trace thing` rather than defining new ways to do `traceWith`.

Reply:
Yeah, the idea is that you construct your `Trace` in some way, and then use `traceWith` as the one true way of "interpreting" it to get an effect. I think this leads to a more modular/composable feel.

Reply:
i.e. it's better to give a way to construct a new `Trace` than to give a way to construct a new `m ()` from a `Trace m t`, because `m ()` is too general.